### PR TITLE
object-based logging for Python

### PR DESCRIPTION
**Pull request by billschereriii**
Enable user-initiated logging from {client, dataset, logcontext}.{log_data, log_warning, log_error} methods for Python users

### CODE REVIEW

**Review by MattToast — CHANGES_REQUESTED** [review]
Looks great!! One small concern regarding multiple pointers that look like they should always be pointing to the same object that I wanted bring to attention but otherwise looks fantastic!

File: src/python/module/smartredis/srobject.py [hunk 1]
```diff
@@ -38,19 +38,17 @@ def __init__(self, context):
...
         """Initialize a SRObject object from
         a PySRObject object
 
-        :param srobject: The pybind PySRObject object
-                           to use for construction
+        :param srobject: The pybind PySRObject object to use for construction
         :type srobject: PySRObject
-        :return: The newly constructor SRObject from
-                 the PySRObject
+        :return: The newly constructor SRObject from the PySRObject
```
[the remark this answers is not in the document]
Comment (billschereriii) [reply src/python/module/smartredis/srobject.py:51]: fixed

**Review by MattToast — APPROVED** [review]
LGTM!!

**Review by ashao — APPROVED** [review]
Looks great, had a similar comment to @MattToast, but I like his suggestion better.

**Pull request merged**